### PR TITLE
Classify and plot

### PR DESCRIPTION
**Pull request by wcornwell**
adding prediction and a bit of plotting functionality

### CODE REVIEW

File: R/classify.R [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ggplot2::ggplot(ggplot2::aes(x = window, y = mode_behavior, color = mode_behavior)) +
+    ggplot2::geom_point(size = 3) +
+    ggplot2::labs(
+      x = "Time",
+      y = "Mode Behavior",
+      title = paste0("Mode Behavior Every ", window_minutes, " Minutes")
+    ) +
+    ggplot2::theme_minimal() -> plotting
+  print(plotting)
+  return(mode_sum)
```
Comment (dfalster) [inline R/classify.R:27]: For the return, I wonder if if should return both data and the plot? What if you want to edit the plot? 

Could also be invisible?

File: R/classify.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#' @param dat A data frame containing the data.
+#' @param MSOM_path classifier
+#' @param quiet chill out
+#' @return A data frame with a column of behaviors
+#' @rdname classify_behaviors
+#' @export
+#'
+classify_behaviors <- function(
+                               dat,
+                               MSOM_path = "tests/testthat/MSOM_8by7.rda",
```
Comment (dfalster) [inline R/classify.R]: This will break when used in package. So better make it available as package data.

File: R/classify.R [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                               quiet = FALSE) {
+  #dat <- moving_window_calcs_2(acc_dat) #this might move out of the function
+  dd <- as.matrix(dat[, -1])
+  load(file = MSOM_path)
+  ssom.pred <- kohonen:::predict.kohonen(MSOM, newdata = dd, whatmap = 1) # this is slow part
```
Comment (dfalster) [inline R/classify.R]: perhaps add a seperate issue to see if can speed it up? Or at least profile.

**Review by dfalster — CHANGES_REQUESTED** [review]
Look promising. The model I(MSOM_8by7.rda) s 13MB, so before adding to repo I'd like to know if really needs to be that big. 

Also, do we want figures saved in the repo?

File: R/classify.R (hunk 2 above)
Comment (dfalster) [reply R/classify.R]: But also, see comment above about file size

File: R/classify.R (hunk 1 above)
Comment (wcornwell) [reply R/classify.R:27]: should probably be two functions to do it properly?

File: R/classify.R (hunk 2 above)
Comment (wcornwell) [reply R/classify.R]: > make it available as package data

I forgot how to do this

File: R/classify.R (hunk 3 above)
Comment (wcornwell) [reply R/classify.R]: adding issue

File: R/classify.R (hunk 2 above)
Comment (wcornwell) [reply R/classify.R]: i think i once knew, but can't remember

**Comment by wcornwell** [comment]
adding figs to the readme, following: 
```
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
``` but could go elsewhere?

File: R/classify.R (hunk 1 above)
Comment (wcornwell) [reply R/classify.R:27]: plotting is so trivial, pulling out for now

File: R/classify.R (hunk 2 above)
Comment (wcornwell) [reply R/classify.R]: there is something in the object called "whatmap" that related to the data and looks like it's necessary...not simple...

Comment (dfalster) [reply R/classify.R]: I have a solution to filesize, will push later today or tomorrow. Off to EERC workshop

**Review by dfalster — APPROVED** [review]

**Pull request merged**